job.c

```diff
@@ -72,6 +72,40 @@ rehash()
     }
 }
 
+static void
+rehash_down()
```

Comment:
This function is almost identical to rehash, it just chooses a different next table size (cur_prime-1 instead of cur_prime+1). Consider parameterizing the existing function instead of copy-pasting it.